storybook/src/components/modal/modal.core-stories.ts

```diff
@@ -0,0 +1,48 @@
+import type { Meta } from "@storybook/web-components";
+import { ModalArgs, modalMeta, modalStories } from "dso-toolkit";
+
+import { templateContainer } from "../../templates";
+
+import { decorator } from "./modal.decorator";
+
+import readme from "@dso-toolkit/core/src/components/modal/readme.md?raw";
+
+import {
+  activeBody,
+  activeFooter,
+  confirmBody,
+  confirmFooter,
+  passiveBody,
+  passiveFooter,
+  loadingBody,
+  datePickerBody,
+} from "./modal.content";
+
+const meta: Meta<ModalArgs> = {
+  ...modalMeta({ readme }),
+  title: "Core/Modal",
+};
+
+export default meta;
+
+const { Passive, Confirm, Fullscreen, Active, WithDatepicker, Loading } = modalStories({
+  templateContainer,
+  storyTemplates: (templates) => {
+    const { modalTemplate } = templates;
+
+    return {
+      modalTemplate,
+      activeBody: activeBody(),
+      activeFooter: activeFooter(templates),
+      passiveBody: passiveBody(templates),
+      passiveFooter: passiveFooter(templates),
+      confirmBody: confirmBody(templates),
+      confirmFooter: confirmFooter(templates),
+      loadingBody: loadingBody(templates),
+      datePickerBody: datePickerBody(templates),
+    };
+  },
+  decorator,
```

Comment:
De decorator is alleen nodig voor het HTML/CSS component omdat het HTML/CSS component extra implementatie details voor de afnemer nodig heeft. De decorator vult dat gemis op.

Het Web Component is echter volledig zelfvoorzienend. Heb je deze decorator toegevoegd omdat iets niet werkte of omdat je zag dat de verschillende implementaties uit elkaar liepen?

Reply:
Ik ga er naar kijken. Kreeg het optionele aspect van de decorator niet voor elkaar. Ga het aanpassen.

Reply:
Aangepast